zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/AQLGenerator.scala

```diff
@@ -121,9 +135,9 @@ class AQLGenerator extends IQLGenerator {
         """.stripMargin
       )
       //TODO check if the vars are duplicated
-      val field: Field = ??? // get field from lookup table
+      val fieldMap: Map[String, Field] = schemaMap.flatMap(_._2.fieldMap) // get fields from lookup and query table
       producedVar ++= lookup.as.zip(lookup.selectValues).map { p =>
-        p._1 -> AQLVar(new Field(p._1, field.dataType), s"$lookupVar.${p._2}")
+        p._1 -> AQLVar(new Field(p._1, fieldMap(p._2).dataType), s"$lookupVar.${p._2}")
```

Comment:
using `p` as a pair is always hard to understand what P actually is. I suggest to use `case match` as following

```
... map { case ( reName: String, oriName: String) =>
   reName -> ***** oriName ***
}
```
Then we will know what is `_1` and what is `_2`.